docs/design/yarn-v1.md

```diff
@@ -0,0 +1,169 @@
+
+    yarn install --non-interactive --frozen-lockfile --disable-pnp --no-default-rc --ignore-engines
+
+Command: [yarn install](https://classic.yarnpkg.com/en/docs/cli/install)
+
+Options:
+ - non-interactive: Disable interactive prompts
+ - frozen-lockfile: Fail if yarn.lock updates are needed
+ - disable-pnp: Disable Plug'n'Play installation
+ - no-default-rc: Prevent Yarn from automatically detecting .yarnrc and .npmrc files
+ - ignore-engines: Ignore the engine node version during the prefetch
+
+ The behavior of Yarn can also be configured using select environment variables. The following ones should be set for the prefetch:
+
+    COREPACK_ENABLE_DOWNLOAD_PROMPT=0
+    COREPACK_ENABLE_PROJECT_SPEC=0 (If we decide to use the global default)
+    YARN_YARN_OFFLINE_MIRROR=<absolute path to the request output directory>
+    YARN_YARN_OFFLINE_MIRROR_PRUNING=false
+    YARN_IGNORE_PATH=true (Use the corepack yarn version, not any Yarn version specified by the user)
+    YARN_IGNORE_SCRIPTS=true
+
+### Build Configuration
+The user build can be configured to use the offline mirror by configuring the following environment variables:
+
+    YARN_YARN_OFFLINE_MIRROR=<request output directory>
+    YARN_YARN_OFFLINE_MIRROR_PRUNING=false
+
+## High-Level Procedure
+ - Since we're executing Yarn commands directly, process the request in an [isolated temporary directory](https://github.com/containerbuildsystem/cachi2/blob/6953607b6ef52fd3f0bef7059d2c926767b1022b/cachi2/core/resolver.py#L41) to avoid any potential changes to the repository
+ - Determine whether to process the request with yarn v1 or yarn v2+
+ - Read the project files
+   - Read/Require package.json
+   - Read/Require yarn.lock
+     - Ensure yarn.lock is formatted for Yarn 1.x
+ - Verify the repository
+   - Ensure node_modules directory is not present
+ - Gather the SBOM Components
+   - Get the main package component data from package.json
+   - Get the workspaces from package.json and add a component for each workspace according to the workspace package.json data
+   - Parse yarn.lock with pyarn and add components for each
+     - Report missing integrity keys from yarn.lock as missing hashes
+     - Ensure that there are no collisions between expected archive names in the offline mirror
+ - Configure environment for the prefetch
+   - Ensure offline mirror is used
+   - Ensure offline mirror is not automatically pruned
+   - Ensure package scripts are ignored
+   - Ensure Yarn 1.x is used via corepack
+ - Do the prefetch
+   - Run yarn install
+ - Configure environment for the user build
+   - Ensure offline mirror is used from the request output
+   - Ensure offline mirror is not pruned automatically
+
+## Potential Follow-Ups
+ - Allow versions of yarn other than the latest 1.x release to be specified by the user
+ - Investigate allowing more user-specified settings during the prefetch via .yarnrc/.npmrc
```

Comment:
Side note, I don't think we'll want to record this in the final design doc when the template materializes as I think the point of ADRs/RFCs/Blueprints is to capture a snapshot in a given time and provide updates in a form of new docs when needed.

Reply:
I agree, but I do see value in clearly noting these as out-of-scope in the initial implementation. I'd probably move these to the proposed "decison/outcome" section of the new design doc format as limitations rather than follow-ups. How does that sound?

Reply:
FWIW, I renamed the similar section in the Bundler design to "out of scope".

Reply:
Seems to be generally accepted, works for me.